src/host.rs

```diff
@@ -388,4 +430,68 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn socket_read() {
```

Comment:
Can we test what follows at the TcpStream level instead?

Reply:
pretty much the same issue as in https://github.com/tokio-rs/turmoil/pull/61#discussion_r1062734834
What exactly can we test at a higher level that is not already covered by the test we have, given that the only thing we can really do is wrap the poll methods in a `poll_fn`?